docs/en/sql-reference/functions/bit-functions.md

```diff
@@ -117,6 +117,46 @@ Result:
 
 ## bitRotateRight(a, b) {#bitrotaterighta-b}
 
+## bitSlice(s, offset, length)
+
+Returns a substring starting with the bit from the ‘offset’ index that is ‘length’ bits long. bits indexing starts from
+1
+
+**Syntax**
+
+``` sql
+bitSlice(s, offset[, length])
+```
+
+**Arguments**
+
+- `s` — s is [String](../../sql-reference/data-types/string.md)
+  or [FixedString](../../sql-reference/data-types/fixedstring.md).
+- `offset` — The start index with bit, A positive value indicates an offset on the left, and a negative value is an
+  indent on the right. Numbering of the bits begins with 1.
+- `length` — The length of substring with bit. If you specify a negative value, the function returns an open substring [
+  offset, array_length - length). If you omit the value, the function returns the substring [offset, the_end_string]
+
+**Returned value**
+
+- The substring. [String](../../sql-reference/data-types/string.md)
+
+**Example**
+
+Query:
+
+``` sql
+select bin('Hello'), bin(bitSlice('Hello', 1, 8))
+```
+
+Result:
+
+``` text
+┌─bin('Hello')─────────────────────────────┬─bin(bitSlice('Hello', 1, 8))─┐
+│ 0100100001100101011011000110110001101111 │ 01001000                     │
```

Comment:
The minimum unit of string is byte, so we add 0 to the right and fill it with 1byte. I don't know if there is a better way. 
> why right, not left?  

I think most usually expectations for return are valid data at the beginning
for example
```
 select bin(bitSlice(unbin('1111111111111111'), 1, 9));

//right
┌─bin(bitSlice(unbin('1111111111111111'), 1, 9))─┐
│ 1111111110000000                               │
└────────────────────────────────────────────────┘


//left
┌─bin(bitSlice(unbin('1111111111111111'), 1, 9))─┐
│ 0000000111111111                               │
└────────────────────────────────────────────────┘


```
I think add zero pad at right will reduce its effect. 

Or we specify that the length must be a multiple of 8, But it will bring some restrictions on its use.

> Also what we will do when slice out of bound.  

```
SELECT
    number,
    bin('' AS e),
    bin(bitSlice(e, number, number + 1))
FROM numbers(10)

┌─number─┬─bin('')─┬─bin(bitSlice('', number, plus(number, 1)))─┐
│      0 │         │                                            │
│      1 │         │                                            │
│      2 │         │                                            │
│      3 │         │                                            │
│      4 │         │                                            │
│      5 │         │                                            │
│      6 │         │                                            │
│      7 │         │                                            │
│      8 │         │                                            │
│      9 │         │                                            │
└────────┴─────────┴────────────────────────────────────────────┘

```
The test is normal here. I try to find out why